src/journal/journalctl.c

```diff
@@ -1460,7 +1460,7 @@ static int add_boot(sd_journal *j) {
         r = get_boots(j, NULL, &boot_id, arg_boot_offset);
         assert(r <= 1);
         if (r <= 0) {
-                const char *reason = (r == 0) ? "No such boot ID in journal" : strerror_safe(r);
+                const char *reason = STRERROR_OR_EOF(r, "No such boot ID in journal");
```

Comment:
hmm, this is a different case. I wouldn't subsume this under STRERROR_OR_EOF()

Reply:
Fixed.